### ISSUE TITLE
Appium - weird issue when trying to do send key in iOS devices in safari for web application

### ISSUE THREAD

**Issue by aadityaawasthi** [appium/appium#13238]
I am facing weird issue when trying to do send key in safari for iOS devices using Appium .

I am using bellow tools for mobile automation:

Appium Version : 1.13.0 iPhone Simulator: 7 (Even Tried all other also) Xcode : Version 10.3 (10G8)

Issue:

its login page where I need to enter userId and Password when I enter both input then it will enable Login button. where Password field is encrypted and have "Show" button.

I am using bellow code with that I am able to set valued in userId and password but

its not enabling Login button (not application issue) when same time I am deleting one value from username and enter manually from keyboard its immediately enabling that Login button.
Password is encrypted and shown as "*******" but when entering from APPIUM using send key then its entering but displaying blank "when click on show button its showing that value " and when I am trying to delete that manually from keyboard then I don't know why but its also deleting username.

I have tried tap, javascript Executer. Even I tried to enter value from soft keyboard but its not showing in iOS devices even changed from "Hardware > Keyboard > Toggle Software Keyboard" and also tried other options.

WebElement username = driver.findElement(By.xpath("//*[@id='username']"));
WebElement password = driver.findElement(By.xpath("//*[@id='password-input']"));
username.click();
Thread.sleep(5000);
username.sendKeys("testUsername");
password.click();
password.sendKeys("testUsername");

<img width="347" alt="Screen Shot 2019-09-15 at 12 01 28 AM" src="https://user-images.githubusercontent.com/16005005/64916597-436abb80-d74e-11e9-82cd-116f450f8b02.png">

<img width="347" alt="Screen Shot 2019-09-15 at 12 01 46 AM" src="https://user-images.githubusercontent.com/16005005/64916598-4a91c980-d74e-11e9-9ad1-212b838ca97f.png">

**Comment by mykola-mokhnach** [comment]
It might be that your web app expects some specific event to happen in order to trigger the action on the button. Please check it with your developers or add more logging on the app itself to figure out how the manually generated events sequence differs from the automated one.

**Comment by aadityaawasthi** [comment]
Thanks @mykola-mokhnach  for your response.

I don't think its application issue because I ran the same in android and its working fine even when in android I am clicking on input field then its opening android keyboard to enter input.
I thought its Safari issue but when I ran script using safari Webdriver in iOS (not in mobile) it worked  fine.
when I am trying to enter value using iOS devices (iPhone,ipad) then its not even opening keyboard to enter values.

and also can I know how I can find manually generated events sequence differs from the automated one.

Appreciate your help.
